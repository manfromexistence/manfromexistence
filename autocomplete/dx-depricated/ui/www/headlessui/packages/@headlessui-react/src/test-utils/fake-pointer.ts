export class FakePointer {
  private x: number = 0
  private y: number = 0

  constructor(
    private width: number,
    private height: number
  ) {
    this.width = width
    this.height = height
  }

  get options() {
    return {
      screenX: this.x,
      screenY: this.y,
    }
  }

  randomize() {
    this.x = Math.floor(Math.random() * this.width)
    this.y = Math.floor(Math.random() * this.height)
  }

  advance(amount: number = 1) {
    this.x += amount

    if (this.x >= this.width) {
      this.x %= this.width
      this.y++
    }

    if (this.y >= this.height) {
      this.y %= this.height
    }
  }

  /**
   * JSDOM does not support pointer events.
   * Because of this when we try to set the pointer position it returns undefined so our checks fail.
   *
   * This runs the callback with the TEST_IGNORE_TRACKED_POINTER environment variable set to 1 so we bypass the checks.
   */
  bypassingTrackingChecks(callback: () => void) {
    let original = process.env.TEST_BYPASS_TRACKED_POINTER
    process.env.TEST_BYPASS_TRACKED_POINTER = '1'
    callback()
    process.env.TEST_BYPASS_TRACKED_POINTER = original
  }
}

/**
 * A global pointer for use in pointer and mouse event checks
 */
export let pointer = new FakePointer(1920, 1080)
