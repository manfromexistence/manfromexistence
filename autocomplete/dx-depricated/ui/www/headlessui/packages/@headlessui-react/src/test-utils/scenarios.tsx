import { render, screen } from '@testing-library/react'
import React from 'react'
import { Description, Field, Fieldset, Label } from '..'
import {
  assertActiveElement,
  assertDisabledish,
  assertLinkedWithDescription,
  assertLinkedWithLabel,
  getControl,
  getDescriptions,
  getLabel,
  getLabels,
} from './accessibility-assertions'
import { click } from './interactions'
import { suppressConsoleLogs } from './suppress-console-logs'

export function commonControlScenarios(Control: React.ComponentType<any>) {
  describe('Rendering composition', () => {
    describe('Inside `Field`', () => {
      it('should mark the control as disabled, if the `Field` is disabled', () => {
        render(
          <Field disabled>
            <Control />
          </Field>
        )

        assertDisabledish(getControl())
      })

      it('should link a control and a `Label` when inside a `Field`', () => {
        render(
          <Field>
            <Label>My Label</Label>
            <Control />
          </Field>
        )

        assertLinkedWithLabel(getControl(), getLabels())
      })

      it('should link a control and multiple `Label` components when inside a `Field`', () => {
        render(
          <Field>
            <Label>My Label #1</Label>
            <Label>My Label #2</Label>
            <Control />
          </Field>
        )

        assertLinkedWithLabel(getControl(), getLabels())
      })

      it('should link a control and a `Description` when inside a `Field`', () => {
        render(
          <Field>
            <Control />
            <Description>My Description</Description>
          </Field>
        )

        assertLinkedWithDescription(getControl(), getDescriptions())
      })

      it('should link a control and multiple `Description` components when inside a `Field`', () => {
        render(
          <Field>
            <Control />
            <Description>My Description #1</Description>
            <Description>My Description #2</Description>
            <Description>My Description #3</Description>
          </Field>
        )

        assertLinkedWithDescription(getControl(), getDescriptions())
      })

      it('should link a control with a `Label` and a `Description` when inside a `Field`', () => {
        render(
          <Field>
            <Label>My Label</Label>
            <Control />
            <Description>My Description</Description>
          </Field>
        )

        assertLinkedWithDescription(getControl(), getDescriptions())
        assertLinkedWithLabel(getControl(), getLabels())
      })
    })
  })

  describe('Mouse interactions', () => {
    describe('Inside `Field`', () => {
      it('should be possible to click a `Label`, and focus the control when in a `Field`', async () => {
        render(
          <Field>
            <Label>My Label</Label>
            <Control />
          </Field>
        )

        assertActiveElement(document.body)
        await click(getLabel())
        assertActiveElement(getControl())
      })

      it('should not be possible to click a `Label`, if the `Label` has the `passive` prop', async () => {
        render(
          <Field>
            <Label passive>My Label</Label>
            <Control />
          </Field>
        )

        assertActiveElement(document.body)
        await click(getLabel())
        assertActiveElement(document.body)
      })

      it('should not be possible to click a `Label` and focus the control, if the control is disabled', async () => {
        render(
          <Field>
            <Label>My Label</Label>
            <Control disabled />
          </Field>
        )

        assertActiveElement(document.body)
        await click(getLabel())
        assertActiveElement(document.body)
      })

      it('should not be possible to click a `Label` and focus the control, if the `Field` is disabled', async () => {
        render(
          <Field disabled>
            <Label>My Label</Label>
            <Control />
          </Field>
        )

        assertActiveElement(document.body)
        await click(getLabel())
        assertActiveElement(document.body)
      })
    })

    describe('Inside `Fieldset`', () => {
      it('should not be possible to click a `Label` and focus the control, if the `Fieldset` is disabled', async () => {
        render(
          <Fieldset disabled>
            <Field>
              <Label>My Label</Label>
              <Control />
            </Field>
          </Fieldset>
        )

        assertActiveElement(document.body)
        await click(getLabel())
        assertActiveElement(document.body)
      })
    })
  })
}

export function commonFormScenarios(
  Control: React.ComponentType<any>,
  {
    performUserInteraction,
  }: { performUserInteraction: (control: HTMLElement | null) => PromiseLike<void> }
) {
  describe('Form compatibility', () => {
    it('should render native (hidden) form elements for the control', () => {
      render(
        <form>
          <Control name="foo" />
        </form>
      )

      expect(document.querySelector('[name=foo]')).toBeInTheDocument()
    })

    it('should submit the form with all the data', async () => {
      let formDataMock = jest.fn()

      render(
        <form
          onSubmit={(e) => {
            e.preventDefault()
            formDataMock(new FormData(e.target as HTMLFormElement))
          }}
        >
          <Control name="foo" />
          <button>Submit</button>
        </form>
      )

      // Submit form
      await click(screen.getByText('Submit'))

      // Ensure the form was submitted with the `foo` input present
      expect(formDataMock.mock.calls[0][0].has('foo')).toBe(true)
    })

    it('should not submit the data if the control is disabled', async () => {
      let submits = jest.fn()

      function Example() {
        return (
          <form
            onSubmit={(event) => {
              event.preventDefault()
              submits([...new FormData(event.currentTarget).entries()])
            }}
          >
            <input type="hidden" name="foo" value="bar" />
            <Control name="bar" disabled />
            <button>Submit</button>
          </form>
        )
      }

      render(<Example />)

      // Submit the form
      await click(screen.getByText('Submit'))

      // Verify that the form has been submitted
      expect(submits).toHaveBeenLastCalledWith([
        ['foo', 'bar'], // The only available field
      ])
    })

    it(
      'should reset the control when the form is reset',
      suppressConsoleLogs(async () => {
        let formDataMock = jest.fn()

        render(
          <form
            onSubmit={(e) => {
              e.preventDefault()
              formDataMock(new FormData(e.target as HTMLFormElement))
            }}
          >
            <Field>
              <Label>The Label</Label>
              <Control name="foo" />
            </Field>

            <button>Submit</button>
            <button type="reset">Reset</button>
          </form>
        )

        // Submit the form to get the initial state of the form
        await click(screen.getByText('Submit'))
        let formState = Object.fromEntries(formDataMock.mock.calls[0][0])

        // Make changes to the control
        await performUserInteraction(getControl())

        // Submit form
        await click(screen.getByText('Submit'))

        // Ensure the form was, and the values are different
        let newFormState = Object.fromEntries(formDataMock.mock.calls[1][0])
        expect(newFormState).not.toEqual(formState)

        // Reset the form
        await click(screen.getByText('Reset'))

        // Ensure the form was reset
        await click(screen.getByText('Submit'))

        // Ensure the form state looks like the initial state
        let resetFormState = Object.fromEntries(formDataMock.mock.calls[2][0])
        expect(resetFormState).toEqual(formState)
      })
    )
  })
}

export function commonRenderingScenarios(
  Control: React.ComponentType<any>,
  { getElement }: { getElement: () => HTMLElement | null }
) {
  describe('Rendering', () => {
    it('should render a control', async () => {
      render(<Control />)

      expect(getElement()).toBeInTheDocument()
    })

    it('should have an `id` attached', () => {
      render(<Control />)

      expect(getElement()).toHaveAttribute('id')
    })

    it('should be possible to override the `id`', () => {
      render(<Control id="foo" />)

      expect(getElement()).toHaveAttribute('id', 'foo')
    })
  })
}
