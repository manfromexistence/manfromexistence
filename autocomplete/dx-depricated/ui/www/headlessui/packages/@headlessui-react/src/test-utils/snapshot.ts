import { useLayoutEffect, useRef, type MutableRefObject } from 'react'

export function createSnapshot() {
  let snapshots: HTMLElement[] = []

  return {
    get latest() {
      return snapshots.at(-1)
    },

    get firstChild() {
      return this.latest?.firstChild ?? null
    },

    use() {
      let ref = useRef<HTMLElement>(null)
      useLayoutEffect(() => this.take(ref), [])
      return ref
    },

    take(ref: MutableRefObject<HTMLElement | null>) {
      snapshots.push(ref.current!.parentElement!.cloneNode(true) as any)
    },
  }
}
