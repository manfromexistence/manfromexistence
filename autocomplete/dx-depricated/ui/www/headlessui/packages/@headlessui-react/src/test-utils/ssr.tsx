import {
  cleanup,
  render,
  screen,
  type RenderOptions,
  type RenderResult,
} from '@testing-library/react'
import React, { type ReactElement } from 'react'
import { renderToString } from 'react-dom/server'
import { env } from '../utils/env'

type ServerRenderOptions = Omit<RenderOptions, 'queries'> & {
  strict?: boolean
}

interface ServerRenderResult {
  type: 'ssr' | 'hydrate'
  contents: string
  result: RenderResult
  hydrate: () => Promise<ServerRenderResult>
}

export async function renderSSR(
  ui: ReactElement,
  options: ServerRenderOptions = {}
): Promise<ServerRenderResult> {
  let container = document.createElement('div')
  document.body.appendChild(container)
  options = { ...options, container }

  if (options.strict) {
    options = {
      ...options,
      wrapper({ children }) {
        return <React.StrictMode>{children}</React.StrictMode>
      },
    }
  }

  env.set('server')
  let contents = renderToString(ui)
  let result = render(<div dangerouslySetInnerHTML={{ __html: contents }} />, options)

  async function hydrate(): Promise<ServerRenderResult> {
    cleanup()

    container.remove()

    container = document.createElement('div')
    container.innerHTML = contents
    document.body.appendChild(container)

    env.set('client')

    let newResult = render(ui, {
      ...options,
      container,
      hydrate: true,
    })

    return {
      type: 'hydrate',
      contents: container.innerHTML,
      result: newResult,
      hydrate,
    }
  }

  return {
    type: 'ssr',
    contents,
    result,
    hydrate,
  }
}

export async function renderHydrate(el: ReactElement, options: ServerRenderOptions = {}) {
  return renderSSR(el, options).then((r) => r.hydrate())
}

export { screen }
