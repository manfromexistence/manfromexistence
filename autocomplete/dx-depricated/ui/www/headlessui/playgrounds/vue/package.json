{
  "name": "playground-vue",
  "private": true,
  "version": "0.0.0",
  "directories": {
    "example": "examples"
  },
  "scripts": {
    "prebuild": "npm run build --workspace=@headlessui/vue && npm run build --workspace=@headlessui/tailwindcss",
    "predev": "npm run build --workspace=@headlessui/vue && npm run build --workspace=@headlessui/tailwindcss",
    "dev:tailwindcss": "npm run watch --workspace=@headlessui/tailwindcss",
    "dev:headlessui": "npm run watch --workspace=@headlessui/vue",
    "dev:next": "vite serve",
    "dev": "npm-run-all -p dev:*",
    "build": "NODE_ENV=production vite build",
    "lint-types": "echo",
    "clean": "rimraf ./dist"
  },
  "dependencies": {
    "@headlessui/vue": "*",
    "@heroicons/vue": "^1.0.6",
    "@tailwindcss/forms": "^0.5.2",
    "@tailwindcss/typography": "^0.5.2",
    "autoprefixer": "^10.4.7",
    "postcss": "^8.4.14",
    "tailwindcss": "^3.3.3",
    "vue": "^3.2.27",
    "vue-flatpickr-component": "^9.0.5",
    "vue-router": "^4.0.0"
  },
  "devDependencies": {
    "@floating-ui/vue": "^1.0.2",
    "@vitejs/plugin-vue": "^3.0.3",
    "vite": "^3.0.0"
  }
}
