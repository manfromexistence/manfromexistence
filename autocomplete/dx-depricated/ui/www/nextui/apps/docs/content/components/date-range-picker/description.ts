const App = `import {DateRangePicker} from "@nextui-org/react";

export default function App() {
  return (
    <DateRangePicker
      label="Stay duration"
      description="Please enter your stay duration"
      className="max-w-xs"
    />
  );
}`;

const react = {
  "/App.jsx": App,
};

export default {
  ...react,
};
