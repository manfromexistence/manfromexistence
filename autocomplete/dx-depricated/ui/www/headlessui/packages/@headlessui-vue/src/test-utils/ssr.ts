import { createApp, createSSRApp, nextTick } from 'vue'
import { renderToString } from 'vue/server-renderer'
import { env } from '../utils/env'

export async function renderSSR(component: any, rootProps: any = {}) {
  let container = document.createElement('div')
  document.body.appendChild(container)

  // Render on the server
  env.set('server')
  let app = createSSRApp(component, rootProps)
  let contents = await renderToString(app)
  container.innerHTML = contents

  return {
    contents,
    async hydrate() {
      let app = createApp(component, rootProps)
      app.mount(container)

      await nextTick()

      return {
        contents: container.innerHTML,
      }
    },
  }
}

export async function renderHydrate(component: any, rootProps: any = {}) {
  return renderSSR(component, rootProps).then(({ hydrate }) => hydrate())
}
