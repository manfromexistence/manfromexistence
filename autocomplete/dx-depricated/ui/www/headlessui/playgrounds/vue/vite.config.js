import vue from '@vitejs/plugin-vue'
import { defineConfig } from 'vite'

export default defineConfig({
  server: { port: 3000 },
  plugins: [vue()],
})
