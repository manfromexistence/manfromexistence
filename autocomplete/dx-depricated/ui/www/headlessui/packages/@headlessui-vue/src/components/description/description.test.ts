import { defineComponent, h, nextTick, ref } from 'vue'
import { getByText } from '../../test-utils/accessibility-assertions'
import { click } from '../../test-utils/interactions'
import { render } from '../../test-utils/vue-testing-library'
import { Description, useDescriptions } from './description'

function format(input: Element | null | string) {
  if (input === null) throw new Error('input is null')
  let contents = (typeof input === 'string' ? input : (input as HTMLElement).outerHTML).trim()
  return contents
}

jest.mock('../../hooks/use-id')

beforeAll(() => {
  jest.spyOn(window, 'requestAnimationFrame').mockImplementation(setImmediate as any)
  jest.spyOn(window, 'cancelAnimationFrame').mockImplementation(clearImmediate as any)
})

afterAll(() => jest.restoreAllMocks())

it('should be possible to use useDescriptions without using a Description', async () => {
  let { container } = render(
    defineComponent({
      components: { Description },
      setup() {
        let describedby = useDescriptions()

        return () =>
          h('div', [h('div', { 'aria-describedby': describedby.value }, ['No description'])])
      },
    })
  )

  expect(format(container.firstElementChild)).toMatchSnapshot()
})

it('should be possible to use useDescriptions and a single Description, and have them linked', async () => {
  let { container } = render(
    defineComponent({
      components: { Description },
      setup() {
        let describedby = useDescriptions()

        return () =>
          h('div', [
            h('div', { 'aria-describedby': describedby.value }, [
              h(Description, () => 'I am a description'),
              h('span', 'Contents'),
            ]),
          ])
      },
    })
  )

  await new Promise<void>(nextTick)

  expect(format(container.firstElementChild)).toMatchSnapshot()
})

it('should be possible to use useDescriptions and multiple Description components, and have them linked', async () => {
  let { container } = render(
    defineComponent({
      components: { Description },
      setup() {
        let describedby = useDescriptions()

        return () =>
          h('div', [
            h('div', { 'aria-describedby': describedby.value }, [
              h(Description, () => 'I am a description'),
              h('span', 'Contents'),
              h(Description, () => 'I am also a description'),
            ]),
          ])
      },
    })
  )

  await new Promise<void>(nextTick)

  expect(format(container.firstElementChild)).toMatchSnapshot()
})

it('should be possible to update a prop from the parent and it should reflect in the Description component', async () => {
  let { container } = render(
    defineComponent({
      components: { Description },
      setup() {
        let count = ref(0)
        let describedby = useDescriptions({ props: { 'data-count': count } })

        return () => {
          return h('div', [
            h('div', { 'aria-describedby': describedby.value }, [
              h(Description, () => 'I am a description'),
              h('button', { onClick: () => count.value++ }, '+1'),
            ]),
          ])
        }
      },
    })
  )

  await new Promise<void>(nextTick)

  expect(format(container.firstElementChild)).toMatchSnapshot()

  await click(getByText('+1'))

  expect(format(container.firstElementChild)).toMatchSnapshot()
})
