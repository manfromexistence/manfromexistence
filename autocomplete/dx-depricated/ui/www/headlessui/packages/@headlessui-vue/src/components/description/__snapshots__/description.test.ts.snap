// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be possible to update a prop from the parent and it should reflect in the Description component 1`] = `"<div><div aria-describedby=\\"headlessui-description-1\\"><p data-count=\\"0\\" id=\\"headlessui-description-1\\">I am a description</p><button>+1</button></div></div>"`;

exports[`should be possible to update a prop from the parent and it should reflect in the Description component 2`] = `"<div><div aria-describedby=\\"headlessui-description-1\\"><p data-count=\\"1\\" id=\\"headlessui-description-1\\">I am a description</p><button>+1</button></div></div>"`;

exports[`should be possible to use useDescriptions and a single Description, and have them linked 1`] = `"<div><div aria-describedby=\\"headlessui-description-1\\"><p id=\\"headlessui-description-1\\">I am a description</p><span>Contents</span></div></div>"`;

exports[`should be possible to use useDescriptions and multiple Description components, and have them linked 1`] = `"<div><div aria-describedby=\\"headlessui-description-1 headlessui-description-2\\"><p id=\\"headlessui-description-1\\">I am a description</p><span>Contents</span><p id=\\"headlessui-description-2\\">I am also a description</p></div></div>"`;

exports[`should be possible to use useDescriptions without using a Description 1`] = `"<div><div>No description</div></div>"`;
